lib/vector-core/buffers/src/lib.rs

```diff
@@ -133,30 +128,39 @@ impl Arbitrary for WhenFull {
     }
 }
 
+/// An item that can be buffered.
+///
+/// This supertrait serves as the base trait for any item that can be pushed into a buffer.
+pub trait Bufferable:
+    ByteSizeOf + EncodeBytes<Self> + DecodeBytes<Self> + Send + Sync + Unpin + Sized + 'static
+{
+}
```

Comment:
Can you walk me through why we need the `'static` lifetime? I don't doubt that the type system is demanding it but I'm surprised. I see you've dropped the generic lifetime off `BufferInputCloner` as well. 

Not opposed, mind, more just needing an education.

Reply:
Cool, makes sense to me.